IMA/BasicIMAPlayer/objc/BasicIMAPlayer/VMAPViewController.m

```diff
@@ -0,0 +1,77 @@
+//
+//  VMAPViewController.m
+//  BasicIMAPlayer
+//
+//  Created by Jeremy Blaker on 10/23/20.
+//  Copyright © 2020 BrightCove. All rights reserved.
+//
+
+#import "VMAPViewController.h"
+
+// See https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/tags for other sample VMAP ad tag URLs
+static NSString * const kVMAPAdTagURL = @"https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&cmsid=496&vid=short_onecue&correlator=";
+
+@interface VMAPViewController ()<BCOVPlaybackControllerDelegate, IMAWebOpenerDelegate>
+
+@end
+
+@implementation VMAPViewController
+
+- (void)setupPlaybackController
+{
+    BCOVPlayerSDKManager *manager = [BCOVPlayerSDKManager sharedManager];
+
+    IMASettings *imaSettings = [[IMASettings alloc] init];
+    imaSettings.ppid = kViewControllerIMAPublisherID;
+    imaSettings.language = kViewControllerIMALanguage;
+
+    IMAAdsRenderingSettings *renderSettings = [[IMAAdsRenderingSettings alloc] init];
+    renderSettings.webOpenerPresentingController = self;
+    renderSettings.webOpenerDelegate = self;
+    
+    // BCOVIMAAdsRequestPolicy provides methods to specify VAST or VMAP/Server Side Ad Rules. Select the appropriate method to select your ads policy.
+    BCOVIMAAdsRequestPolicy *adsRequestPolicy = [BCOVIMAAdsRequestPolicy videoPropertiesVMAPAdTagUrlAdsRequestPolicy];
+    
+    // BCOVIMAPlaybackSessionDelegate defines -willCallIMAAdsLoaderRequestAdsWithRequest:forPosition: which allows us to modify the IMAAdsRequest object
+    // before it is used to load ads.
+    NSDictionary *imaPlaybackSessionOptions = @{ kBCOVIMAOptionIMAPlaybackSessionDelegateKey: self };
+
+    self.playbackController = [manager createIMAPlaybackControllerWithSettings:imaSettings
+                                                          adsRenderingSettings:renderSettings
+                                                              adsRequestPolicy:adsRequestPolicy
+                                                                   adContainer:self.playerView.contentOverlayView
+                                                                viewController:self
+                                                                companionSlots:nil
+                                                                  viewStrategy:nil
+                                                                       options:imaPlaybackSessionOptions];
+    self.playbackController.delegate = self;
+    self.playbackController.autoAdvance = YES;
+    self.playbackController.autoPlay = YES;
+
+    self.playerView.playbackController = self.playbackController;
+
+    // Creating a playback controller based on the above code will create
+    // VMAP / Server Side Ad Rules. These settings are explained in BCOVIMAAdsRequestPolicy.h.
```

Comment:
Question because I'm still learning... does VMAP mean SSAI or can VMAP also be used with CSAI?